internal/armada/server/lease.go

```diff
@@ -176,76 +187,37 @@ func (q *AggregatedQueueServer) StreamingLeaseJobs(stream api.AggregatedQueue_St
 		return errors.WithStack(err)
 	}
 
-	// Return no jobs if we don't have enough work.
-	var res common.ComputeResources = req.Resources
-	if res.AsFloat().IsLessThan(q.schedulingConfig.MinimumResourceToSchedule) {
-		return nil
-	}
-
-	queues, err := q.queueRepository.GetAllQueues()
-	if err != nil {
-		return err
-	}
-
-	activeQueues, err := q.jobRepository.FilterActiveQueues(queue.QueuesToAPI(queues))
-	if err != nil {
-		return err
-	}
-
-	usageReports, err := q.usageRepository.GetClusterUsageReports()
-	if err != nil {
-		return err
-	}
-
 	if err = q.usageRepository.UpdateClusterLeased(&req.ClusterLeasedReport); err != nil {
 		return err
 	}
 
-	// One of the below functions expects a regular lease request,
-	// so we need to convert.
-	leaseRequest := &api.LeaseRequest{
-		ClusterId:           req.ClusterId,
-		Pool:                req.Pool,
-		Resources:           req.Resources,
-		ClusterLeasedReport: req.ClusterLeasedReport,
-		MinimumJobSize:      req.MinimumJobSize,
-		Nodes:               req.Nodes,
-	}
-	nodeResources := scheduling.AggregateNodeTypeAllocations(req.Nodes)
-	clusterSchedulingInfo := scheduling.CreateClusterSchedulingInfoReport(leaseRequest, nodeResources)
-	err = q.schedulingInfoRepository.UpdateClusterSchedulingInfo(clusterSchedulingInfo)
-	if err != nil {
-		return err
+	// Return no jobs if we don't have enough work.
+	var res common.ComputeResources = req.Resources
+	if res.AsFloat().IsLessThan(q.schedulingConfig.MinimumResourceToSchedule) {
+		return nil
 	}
 
-	activeClusterReports := scheduling.FilterActiveClusters(usageReports)
-	activePoolClusterReports := scheduling.FilterPoolClusters(req.Pool, activeClusterReports)
-	activePoolCLusterIds := scheduling.GetClusterReportIds(activePoolClusterReports)
-	clusterPriorities, err := q.usageRepository.GetClusterPriorities(activePoolCLusterIds)
-	if err != nil {
-		return err
-	}
+	// Get jobs to be leased.
+	// TODO: The lease call resource reporting logic should be moved here.
+	var jobs []*api.Job
+	if rand.Float64() < q.schedulingConfig.ProbabilityOfUsingNewScheduler {
+		// New scheduler.
+		// But we need to call the old scheduler with leaseJobs = false because other component rely on its reporting.
+		_, err = q.legacyGetJobs(stream.Context(), req, false)
```

Comment:
what exactly does this do? specifically I'm worried that we already drive redis quite hard, and it looks like we might be doubling the load here?

Reply:
I  checked and  it doesn't seem *too* bad